File: backend/app/schemas/knowledge_card.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class KnowledgeCardItem(BaseModel):
    id: int
    title: str | None = None
    inputType: str
    templateId: str | None = None
    imageNumber: str | None = None
    routeReason: str | None = None
    status: str
    outputImageUrl: str | None = None
    errorMessage: str | None = None
    createdAt: datetime
    updatedAt: datetime


class KnowledgeCardDetail(KnowledgeCardItem):
    inputText: str | None = None
    inputImageUrl: str | None = None
    prompt: str | None = None
    extraPrompt: str | None = None
    difyWorkflowRunId: str | None = None
    difyTaskId: str | None = None
    rawResponse: dict | None = None


class KnowledgeCardListResponse(BaseModel):
    items: list[KnowledgeCardItem]
    total: int
    page: int
    pageSize: int

File: frontend/src/types/knowledgeCard.ts
export interface KnowledgeCardItem {
  id: number;
  title?: string;
  inputType: 'text' | 'image' | 'mixed';
  templateId?: string;
  imageNumber?: string;
  routeReason?: string;
  status: 'processing' | 'succeeded' | 'failed';
  outputImageUrl?: string;
  errorMessage?: string;
  createdAt: string;
  updatedAt: string;
}

export interface KnowledgeCardDetail extends KnowledgeCardItem {
  inputText?: string;
  inputImageUrl?: string;
  prompt?: string;
  extraPrompt?: string;
  difyWorkflowRunId?: string;
  difyTaskId?: string;
  rawResponse?: Record<string, any>;
}

export interface KnowledgeCardListResponse {
  items: KnowledgeCardItem[];
  total: number;
  page: number;
  pageSize: number;
}

export interface KnowledgeCardStreamHandlers {
  onStatus?: (message: string, cardId?: number) => void;
  onWorkflow?: (message: string, node?: string, cardId?: number) => void;
  onPreview?: (imageUrl: string, cardId?: number) => void;
  onDone?: (card: KnowledgeCardDetail) => void;
  onError?: (message: string, cardId?: number) => void;
}
